File: C++/Funciones/isAbundant.cpp
//Función usada en Project Euler #23

bool isAbundant(int n){ //Función para determinar si el número es abundante
     if (n < 12) return false; //No hay números abundantes menores de 12
     else if (n % 2 != 0 ) return false; //No hay números abundantes primos ni impares
     else{
          int divisor = 1, total = 0;
          while (divisor <= (n / 2)){              
              if (n%divisor == 0) total += divisor;              
              divisor += 1;
          }if (total > n) return true;
          else if (total <= n) return false;
     }
}

File: Visual Basic .NET/Funciones/isAbundant.vb
'Función usada en Project Euler #023

Function isAbundant(ByVal n As Integer) As Boolean 'Función para determinar si el número es abundante
    If n < 12 Then
      Return False
    ElseIf isPrime(n) Then
      Return False
    Else
      Dim divisor As Integer = Nothing
      Dim total As Integer = Nothing
      Do While divisor <= (n / 2)
        divisor += 1
        If n Mod divisor = 0 Then
          total += divisor
        End If
    Loop
      If total > n Then
        Return True
      ElseIf total <= n Then
        Return False
      End If
      Return True
    End If
End Function
